scripts/tests/test_validate_viewer_artifacts.py

```
@@ -0,0 +1,56 @@
+        image = validate_viewer_artifacts.validate(manifest)
+
+        self.assertEqual(image, REPO_ROOT / "docs" / "images" / "S3-7-27528-I1-R1.png")
+
+    def test_nonfinite_manifest_number_is_rejected(self) -> None:
+        """NaN must not pass Python validation when Qt rejects the same JSON."""
+        manifest = self._copy_contract()
+        contents = manifest.read_text(encoding="utf-8")
+        self.assertIn('"oversampling": 1.0', contents)
+        manifest.write_text(contents.replace('"oversampling": 1.0', '"oversampling": NaN', 1), encoding="utf-8")
+
+        with self.assertRaisesRegex(validate_viewer_artifacts.ViewerArtifactError, "non-standard JSON constant 'NaN'"):
```

Ruff (0.16.1)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)